### PR TITLE
[7154] Change irods::file_object size to rodsLong_t (4-3-stable)

### PR DESCRIPTION
**Pull request by Nishant2022**
Cherry picked from main

### CODE REVIEW

**Comment by korydraughn** [comment]
Let's add the unit test as a separate commit. That way it's easy to cherry-pick back to the main branch.

File: unit_tests/src/test_file_object.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include <catch2/catch.hpp>
+
+#include "irods/irods_file_object.hpp"
+
+TEST_CASE("file_object")
+{
+    SECTION("test_file_object_size_can_be_negative__issue_7154")
```
Comment (korydraughn) [inline unit_tests/src/test_file_object.cpp:7]: The the C++ unit tests, we don't include underscores to separate words.

Compare your section title to other unit tests and try to make it align with how those are worded.

**Comment by alanking** [comment]
The unit test you wrote is fine and good, but the unit test that I was proposing is less of a unit test in the traditional sense and more of a unit test in the iRODS sense. :) It's more like a C API test. Apologies...

We should try using `rcDataObjOpen` with a `DataObjInp::dataSize` of -1 to make sure that the server does not misbehave like it does when PRC passes a `dataSize` of -1.

File: unit_tests/src/test_file_object.cpp [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 #include "irods/irods_file_object.hpp"
 
-TEST_CASE("file_object")
+TEST_CASE("file object")
 {
-    SECTION("test_file_object_size_can_be_negative__issue_7154")
+    SECTION("test file object size can be negative issue 7154")
```
Comment (korydraughn) [inline unit_tests/src/test_file_object.cpp]: My bad. I was referring to everything except the type.

`file object` -> `file_object`

File: unit_tests/src/test_file_object.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+TEST_CASE("file_object")
+{
+    SECTION("test file object size can be negative issue 7154")
+    {
+        irods::file_object obj{};
+        REQUIRE(obj.size() == 0);
+
+        obj.size(-1);
+        REQUIRE(obj.size() == -1);
```
Comment (alanking) [inline unit_tests/src/test_file_object.cpp:13]: Perhaps we should modify this or add another assertion that `obj.size() < 0` to make sure that there's no funny business with casting...?

File: server/core/include/irods/irods_file_object.hpp [hunk 4]
```diff
@@ -103,7 +104,7 @@ namespace irods {
             virtual int                            l1_desc_idx()     const {
                 return l1_desc_idx_;
             }
-            virtual size_t                         size()            const {
+            virtual rodsLong_t                         size()            const {
```
Comment (trel) [inline server/core/include/irods/irods_file_object.hpp:107]: want to keep these aligned over on the right?

File: server/core/include/irods/irods_file_object.hpp [hunk 5]
```diff
@@ -134,7 +135,7 @@ namespace irods {
             virtual void comm( rsComm_t* _c )                   {
                 comm_            = _c;
             }
-            virtual void size( size_t _v )                       {
+            virtual void size( rodsLong_t _v )                       {
```
Comment (trel) [inline server/core/include/irods/irods_file_object.hpp:138]: same here

**Review by alanking — COMMENTED** [review]
Seems good to me. I vote for a squash, and leave off #s for now.

**Comment by korydraughn** [comment]
Let's squash it without the pounds.

**Comment by korydraughn** [comment]
Given that the file_object header's formatting is not up to par, I think we revert the alignment change in the unit test commit.

That header is going to be formatted in whole at some point in the future.

**Review by alanking — APPROVED** [review]

**Review by korydraughn — APPROVED** [review]
Please cherry-pick the second commit back to main and then add the pounds to this PR.

Nice work.

**Comment by Nishant2022** [comment]
Added pounds

**Pull request merged**